RELEASE/scripts/autoscend/combat/auto_combat_default_stage2.ash

```diff
@@ -234,25 +286,29 @@ string auto_combatDefaultStage2(int round, monster enemy, string text)
 
 		if(canUse($skill[Chest X-Ray]) && equipped_amount($item[Lil\' Doctor&trade; bag]) > 0 && (get_property("_chestXRayUsed").to_int() < 3))
 		{
-			if((my_adventures() < 20) || inAftercore() || (my_daycount() >= 3))
+			if((wantInstaKill || my_adventures() < 20) || inAftercore() || (my_daycount() >= 3))
 			{
 				handleTracker(enemy, $skill[Chest X-Ray], "auto_instakill");
 				loopHandlerDelayAll();
 				return useSkill($skill[Chest X-Ray]);
 			}
 		}
-		if(canUse($skill[shattering punch]) && (get_property("_shatteringPunchUsed").to_int() < 3))
+		if(canUse($skill[shattering punch]) && (get_property("_shatteringPunchUsed").to_int() < 3) && !reserveFreekills)
 		{
-			if((my_adventures() < 20) || inAftercore() || (my_daycount() >= 3))
+			if(!wantInstaKill && my_daycount() == 1 && my_turncount() < 100 && my_adventures() >= 9 && my_mp() < 80)
```

Comment:
should this be wantIntsaKill? Should only prevent punching too soon if we were actually going to punch